src/ui/public/test_harness/test_sharding/get_sharding_params_from_url.js

```diff
@@ -0,0 +1,27 @@
+import { parse as parseUrl } from 'url';
+
+/**
+ *  This function extracts the relevant params from the url and
+ *  returns only the values defined, so that defaults can be
+ *  applied externally
```

Comment:
Maybe this can just be `This function extracts the relevant "shards" and "shard_num" params from the url.` since I would expect any missing query params to be undefined.